pkg/operator/ceph/cluster/mon/mon.go

```diff
@@ -488,20 +488,11 @@ func (c *Cluster) startMon(m *monConfig, hostname string) error {
 	_, err := c.context.Clientset.AppsV1().Deployments(c.Namespace).Create(d)
 	if err != nil {
 		if !errors.IsAlreadyExists(err) {
-			return fmt.Errorf("failed to create mon %s. %+v", m.ResourceName, err)
+			return fmt.Errorf("failed to create mon deployment %s. %+v", m.ResourceName, err)
 		}
-		logger.Debugf("deployment for mon %s already exists. updating if needed", m.ResourceName)
-		p, err := c.context.Clientset.AppsV1().Deployments(c.Namespace).Get(d.Name, metav1.GetOptions{})
-		if err != nil {
-			return fmt.Errorf("failed to update mon deployment %s. failed to inspect preexisting deployment. %+v", d.Name, err)
-		}
-		// Workaround for #2331 targeted for Rook v0.9: only update the deployment if the Rook init
```

Comment:
Was #2331 only applicable to upgrades from 0.8 to 0.9 and now we don't have to worry about that issue? I'm not remembering the details of why multiple upgrades were triggered.

Reply:
From my testing here, the mons aren't totally updated with these label updates. I don't fully recall the reason behind why 2331 was happening ... I think the mon config update may have fixed this.